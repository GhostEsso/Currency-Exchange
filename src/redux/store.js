import { configureStore } from '@reduxjs/toolkit';
import exchangeSlice from './exchangeSlice';

const store = configureStore({
  reducer: {
    exchange: exchangeSlice.reducer,
  },
});

export default store;
