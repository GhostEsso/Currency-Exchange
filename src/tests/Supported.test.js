import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Support from '../components/currency/Supported';

// Import jest-dom for additional matchers
import '@testing-library/jest-dom/';

// Mock axios
jest.mock('axios', () => ({
  get: jest.fn().mockResolvedValue({
    data: {
      supportedCurrenciesMap: {},
    },
  }),
}));

test('renders Support component', () => {
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <Router>
        <Support />
      </Router>
    </Provider>,
  );

  const headingElement = getByText('Currency not found');
  expect(headingElement).toBeInTheDocument();

  // Take a snapshot of the rendered component
  expect(asFragment()).toMatchSnapshot();
});
