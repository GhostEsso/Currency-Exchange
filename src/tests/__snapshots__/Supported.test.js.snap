// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Support component 1`] = `
<DocumentFragment>
  <section
    class="support"
  >
    <div
      class="details"
    >
      <h2>
        Currency not found
      </h2>
    </div>
  </section>
</DocumentFragment>
`;
